app/src/main/java/uk/co/neuralcubes/neuralates/ControlFragment.java

```diff
@@ -308,17 +321,25 @@ public void run() {
     }
 
     @Subscribe
-    public synchronized void updateConcentration(MuseHandler.FocusReading reading) {
-        final int[] rgbColor = mColorMap.iMap(reading.getFocus());
+    public synchronized void updateConcentration(final MuseHandler.FocusReading reading) {
+
         getActivity().runOnUiThread(new Runnable() {
             @Override
             public void run() {
-                mConcentrationBar.setBackgroundColor(Color.rgb(rgbColor[0], rgbColor[1], rgbColor[2]));
-                mConcentrationBar.invalidate();
+              updateConcentrationColor(reading.getFocus());
             }
         });
     }
 
+    void updateConcentrationColor(double value){
+        int[] border = mColorMap.map(0.3);
+        int[] rgbColor = mColorMap.inverseMap(value);
+
+        mConcentrationBar.setBackgroundColor(Color.rgb(rgbColor[0], rgbColor[1], rgbColor[2]));
+        ((FrameLayout)mConcentrationBar.getParent()).setBackgroundColor(Color.rgb(border[0], border[1], border[2]));
```

Comment:
Casting to the least specific type of view needed is a good practice as it won't crash if the XML layout changes in the future. I'd recommend casting to View instead of FrameLayout here.